core/txpool/legacypool/legacypool.go

```diff
@@ -568,9 +572,19 @@ func (pool *LegacyPool) Pending(filter txpool.PendingFilter) map[common.Address]
 				}
 			}
 		}
+		if filter.MaxDATxSize != nil && !pool.locals.contains(addr) {
+			for i, tx := range txs {
+				if types.EstimatedL1SizeScaled(tx.RollupCostData()).Cmp(maxDASizeScaled) > 0 {
+					txs = txs[:i]
+					break
+				}
+			}
+		}
 		if len(txs) > 0 {
 			lazies := make([]*txpool.LazyTransaction, len(txs))
 			for i := 0; i < len(txs); i++ {
+				daBytes := types.EstimatedL1SizeScaled(txs[i].RollupCostData())
```

Comment:
We'd redo the estimation here a 2nd time, right? Given that the estimation is non-trivial as it involves compression, maybe we want to cache all estimates form the above filter loop into a local mapping tx-hash->DA-size?

Reply:
the estimation isn't too costly as the fastlz computation is already cached. do you still think it's worth further caching?

Reply:
oh if it's already cached, then it's fine